src/core/qgscolorramp.h

```diff
@@ -50,6 +50,9 @@ class CORE_EXPORT QgsColorRamp
      */
     virtual QString type() const = 0;
 
+
+    virtual void invert() = 0;
```

Comment:
The color ramps (with the exception of the cpt ramp) currently have almost 100% test coverage. Can you add tests for invert?